### ISSUE TITLE
[BUG] Unable to use automatic signature settings for commit signing

### ISSUE THREAD

**Issue by nan-mu** [k3ii/git-cz#41]
**Describe the bug**
This may be a bug or a feature request. I have used many tools similar to Git cz, and they all read `commit.gpgSign=true` and sign the commit accordingly. However, our tool does not; it even bypasses Git's commit signing check hook. In this sense, it doesn't work as expected.

**To Reproduce**
I am signing with an SSH key. According to [this article](https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key#telling-git-about-your-ssh-key), the signing works correctly when I run `git commit` or `git commit -S` manually.

**Expected behavior**
When I check with `git show/log --show-signature`, if the commit is properly signed, the output should display something like: `Good "git" signature for my@email.com with RSA key SHA256:hash or something?`

**Screenshots**
![image](https://github.com/user-attachments/assets/2302a447-99ec-42c9-af54-435c068953af)

**Environment**
- OS:
   - ![image](https://github.com/user-attachments/assets/7de374ff-af56-4de2-acff-138b4a774c88)

- Version: Installed with `cargo install`, so the version should be v0.1.1.

**Additional context**

None

**Comment by nan-mu** [comment]
Of course, if you are a little busy, I might be able to take a look and submit a related PR this weekend.
And I noticed that the `git-cz` command does not have `--version`

**Comment by k3ii** [comment]
Hello @nan-mu , thank you for opening this issue.

We can support commit signing using [git2-ext](https://github.com/gitext-rs/git2-ext). Or if you know a better way, please submit a PR. 

About the `--version`, we can use the crate `clap` for it.

**Comment by k3ii** [comment]
Here's a [blog](https://blog.hackeriet.no/signing-git-commits-in-rust/) explaining how to do sign commits in Rust.
